android/sdl_android/src/main/java/com/smartdevicelink/util/AndroidTools.java

```diff
@@ -417,4 +421,54 @@ public static void registerReceiver(Context context, BroadcastReceiver receiver,
             }
         }
     }
+
+    /**
+     * A helper method is used to see if this app has permission for UsbAccessory.
+     * We need UsbAccessory permission if we are plugged in via AOA and do not have BLUETOOTH_CONNECT
+     * permission for our service to enter the foreground on Android UPSIDE_DOWN_CAKE and greater
+     * @param context a context that will be used to check the permission.
+     * @return true if connected via AOA and we have UsbAccessory permission.
+     */
+    public static boolean hasUsbAccessoryPermission(Context context) {
+        if (context == null) {
+            return false;
+        }
+        UsbManager manager = (UsbManager) context.getSystemService(Context.USB_SERVICE);
+        if (manager == null || manager.getAccessoryList() == null) {
+            return false;
+        }
+        for (final UsbAccessory usbAccessory : manager.getAccessoryList()) {
+            if (manager.hasPermission(usbAccessory)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Helper method used to check permission passed in.
+     * @param applicationContext Context used to check permission
+     * @param permission String representing permission that is being checked.
+     * @return true if app has permission.
+     */
+    public static boolean checkPermission(Context applicationContext, String permission) {
+        if (applicationContext == null) {
+            return false;
+        }
+        return PackageManager.PERMISSION_GRANTED == ContextCompat.checkSelfPermission(applicationContext, permission);
+    }
```

Comment:
Unless this needs to specifically be the context for the application it should be renamed to just `context`. If it needs to be an application context then any method that calls this one, eg `hasForegroundServiceTypePermission`, would need to make that a requirement. 

```suggestion
    public static boolean checkPermission(Context context, String permission) {
        if (context == null) {
            return false;
        }
        return PackageManager.PERMISSION_GRANTED == ContextCompat.checkSelfPermission(context, permission);
    }
```